src/rf/apps/core/models.py

```diff
@@ -3,6 +3,196 @@
+    to discover similar layers with tags in common.
+    """
+    layer = ForeignKey(Layer)
+    name = CharField(max_length=24)
+
+    def __unicode__(self):
+        return self.name
+
+
+class LayerImage(Model):
+    """
+    Geospatial image uploaded by the user.
+
+    `meta_json` may be populated during geoprocessing and should contain
+    a serialized JSON blob of image metadata.
+    This blob should be in the form of key value pairs (object literal)
+    and will be displayed as-is.
+    """
+    layer = ForeignKey(Layer)
+    source_uri = URLField(max_length=2000)
+    priority = IntegerField(
+        default=0,
+        help_text='The order which images are layered (starting from 0)'
+    )
+    thumb_small = URLField(
+        null=True,
+        blank=True,
+        help_text='80x80 pixels',
+    )
+    thumb_large = URLField(
+        null=True,
+        blank=True,
+        help_text='300x300 pixels',
+    )
+    meta_json = TextField(
+        null=True,
+        blank=True,
+        help_text='Serialized JSON of image metadata',
+    )
+
+    def __unicode__(self):
+        return self.url
+
+
+class LayerMeta(Model):
+    """
+    Immutable state of layer uploading & geoprocessing progress.
+
+    To maintain an audit trail of each status change for a layer, these
+    records should *not* be mutated. Instead, a new record should be created
+    for each status change.
+
+    The data in this table will primarily be maintained by the
+    geoprocessing side of things.
+    """
+    layer = ForeignKey(Layer)
+    state = CharField(max_length=16)
+    error = TextField(null=True, blank=True)
+    thumb_small = URLField(
+        null=True,
+        blank=True,
+        help_text='80x80 pixels',
+    )
+    thumb_large = URLField(
+        null=True,
+        blank=True,
+        help_text='400x150 pixels',
+    )
+    created_at = DateTimeField(auto_now_add=True)
+
+    # TileJSON fields
+    min_zoom = IntegerField(default=0)
+    max_zoom = IntegerField(default=11)
+    bounds = CharField(
+        null=True,
+        max_length=120,
+        help_text='JSON array',
+    )
+    center = CharField(
+        null=True,
+        max_length=60,
+        help_text='JSON array',
+    )
+
+    def __unicode__(self):
+        return self.layer.name
+
+
+class UserProfile(Model):
+    """
+    Additional information for user profiles.
+    """
+    user = ForeignKey(User)
+    organization = CharField(max_length=255, blank=True, default='')
+
+    def __unicode__(self):
+        return self.user.username
+
+
+class UserFavoriteLayer(Model):
+    """
+    Created when a layer is "starred" and destroyed when that layer
+    is "un-starred".
+
+    Users may "star" their own layers or published layers.
+    """
+    user = ForeignKey(User)
+    layer = ForeignKey(Layer)
+    created_at = DateTimeField(auto_now_add=True)
+
+    def __unicode__(self):
+        return self.user.username
```

Comment:
I removed all `__unicode__` methods since they don't get used anywhere, except the one on `Layer`, because it's the only model that directly appears in the Django admin.